src/nvim/ex_getln.c

```diff
@@ -194,6 +194,9 @@ static int cmd_showtail;                /* Only show path tail in lists ? */
 
 static int new_cmdpos;          /* position set by set_cmdline_pos() */
 
+static int extra_char = NUL;    // extra character to display when redrawing
```

Comment:
Should this be a member of `command_line_state` struct?

Reply:
I think we should keep consistent with Vim here to avoid future clashes (even if minimal).

Reply:
There already is `ccline.special_char` (and `special_shift`), though it is only set for external cmdline currently. It could be changed to always set and used for this.

Reply:
Though it could be renamed to `extra_char` for consistency with vim, it must be part of the struct as nvim supports redrawing of nested cmdline unlike vim.

Reply:
Will both new static variables be moved to `command_line_state` or do I simply rename `special_char` and `special_shift` and use those instead? I can add more timer-related patches in the meantime while there is no resolution on this.

Reply:
not `command_line_state` but `struct cmdline_info`, whose static instance `ccline` is the current commandline. In there `special_char` can be renamed `extra_char` for analogy with vim.